### ISSUE TITLE
VC16 build for windows pecl

### ISSUE THREAD

**Issue by N6REJ** [Imagick/imagick#573]
I'm looking for an updated build for https://windows.php.net/downloads/pecl/deps/ImageMagick-7.1.0-18-vc15-x86.zip that uses vc16 & x64.  Where can I find such a thing or create one?

**Comment by terax6669** [comment]
8.2.1 is around the corner and still no build :C I've been waiting for weeks...
I found this: https://wiki.php.net/internals/windows/stepbystepbuild if you manage to build a DLL before one appears on pecl please share.

**Comment by Danack** [comment]
@terax6669 wrote:

> :C

What does ":C" mean? I don't speak emoji.

**Comment by Danack** [comment]
@N6REJ I've actually never been involved in building any extensions for Windows, and I'm not a Windows user.

Reading the front page of https://windows.php.net/ I can see:

> Missing PECL DLLs
17 Oct 2022

You may have already noticed that DLLs for new PECL releases are no longer available for several months. This is because the Windows PECL build machine died, and the team is still working on the long term plan of building DLLs for PECL extensions with a new CI process. We're doing our best to finish that as soon as possible, and keep you up to date.

Which is certainly unfortunate.

I will try to find out what the situation is. For the record, I personally think getting the windows builds available again is something that would be appropriate for the PHP Foundation to be hiring someone to work on full time.

**Comment by N6REJ** [comment]
it would be great if you could expedite it.  We've got every version of imagemick since php 8.0 not working because of it.

**Comment by tmb-github** [comment]
Also, the Imagick windows extension will not work with the current Windows download of XAMPP (8.1.12) (https://www.apachefriends.org/)

**Comment by N6REJ** [comment]
if you want, teach me how to do it and i'll make the imagick ones myself

**Comment by Danack** [comment]
> if you want, teach me how to do it and i'll make the imagick ones myself

I don't know, I've never compiled Imagick on windows.

If anyone wants to write some instructions, I could host them on phpimagick.com

btw, https://mlocati.github.io/articles/php-windows-imagick.html exists and may be of help to some people.

**Comment by tmb-github** [comment]
Sigh, the latest version provided on that page, for 8.1, doesn't work for 8.2. Using PHP 8.2 and following those instructions for 8.1, I get the expected mismatch error:

> [29-Dec-2022 17:35:22 UTC] PHP Warning:  PHP Startup: imagick: Unable to initialize module
> Module compiled with module API=20210902
> PHP    compiled with module API=20220829
> These options need to match
>  in Unknown on line 0

**Comment by WilliamStam** [comment]
thanks @Danack for all your work. waiting patiently for that server to come up hehe

**Comment by terax6669** [comment]
Any update on this?

**Comment by fedot199614** [comment]
Any update on this?

**Comment by ArthurPatriot** [comment]
Hello guys. Trying buid extension on Windows for PHP 8.2 and got this error:
![image](https://user-images.githubusercontent.com/36359548/219075335-8e720a83-2f2a-4467-87fb-4735e21d1a05.png)

If anyone can help, i will share dll here

**Comment by ArthurPatriot** [comment]
Build successful, but get error:
`[15-Feb-2023 16:29:21 UTC] PHP Warning:  PHP Startup: Unable to load dynamic library 'imagick' (tried: d:/openserver/modules/php/PHP_8.2/ext\imagick (The specified module could not be found), d:/openserver/modules/php/PHP_8.2/ext\php_imagick.dll (The specified module could not be found)) in Unknown on line 0`

**Comment by ArthurPatriot** [comment]
Success BUILD!

PHP 8.2.3 x64 TS

[ext-imagick-php8.2-ts-x64.zip](https://github.com/Imagick/imagick/files/10745985/ext-imagick-php8.2-ts-x64.zip)

**Comment by N6REJ** [comment]
Thats fantastic!

**Comment by fedot199614** [comment]
what about nts?

> Success BUILD for Windows! Test on your own risk. Left messages here if it works.
> 
> PHP 8.2.3 x64 TS
> 
> [ext-imagick-php8.2-ts-x64.zip](https://github.com/Imagick/imagick/files/10745985/ext-imagick-php8.2-ts-x64.zip)

greate, but what about nts?

**Comment by ArthurPatriot** [comment]
> 

I had to go through 9 rings of hell to make this build without any instructions or tips... If you really need it, I can make another build, for NTS...)

**Comment by fedot199614** [comment]
> > 
> 
> I had to go through 9 rings of hell to make this build without any instructions or tips... If you really need it, I can make another build, for NTS...) You can download here - [#573 (comment)](https://github.com/Imagick/imagick/issues/573#issuecomment-1431773928)

thanks so much. Will try your build. Also found this great guide and planning use it https://jite.eu/_posts/2021-02-21-imagick-on-php8/

**Comment by tmb-github** [comment]
The thread-safe version (the one I need) will not work on my system (Win11, XAMPP, with PHP 8.2). I suspect it's because the version of ImageMagick I have installed locally (ImageMagick-7.1.0-18-vc15-x64) is different than the version the builds above were compiled against.

@ArthurPatriot, which version of ImageMagick did you compile those builds against?

**Comment by ArthurPatriot** [comment]
> The thread-safe version (the one I need) will not work on my system (Win11, XAMPP, with PHP 8.2). I suspect it's because the version of ImageMagick I have installed locally (ImageMagick-7.1.0-18-vc15-x64/ImageMagick 7.1.0-18 Q16 x64) is different than the version the builds above were compiled against.
> 
> @ArthurPatriot, which version of ImageMagick did you compile those builds against?

All libraries included on zip. You can use latest version of Imagick for Windows.

**Comment by tmb-github** [comment]
@ArthurPatriot Which version of ImageMagick is installed on the machine where you compiled Imagick?

**Comment by ArthurPatriot** [comment]
> @ArthurPatriot Which version of ImageMagick is installed on the machine where you compiled Imagick?

All required libraries exists in zip. 
ImageMagick-7.1.0-62-Q16-HDRI-x64

**Comment by tmb-github** [comment]
@ArthurPatriot Success! Thanks so much!!!

**Comment by pro81723** [comment]
Is this working for Debian Linux? php 8.2

**Comment by ArthurPatriot** [comment]
`sudo add-apt-repository ppa:ondrej/php`
`sudo apt update`
`sudo apt install php8.2-imagick`

**Comment by Rentner2** [comment]
> Success BUILD for Windows! Test on your own risk. Left messages here if it works.
> 
> PHP 8.2.3 x64 TS + NTS
> 
> [ext-imagick-php8.2-ts-x64.zip](https://github.com/Imagick/imagick/files/10745985/ext-imagick-php8.2-ts-x64.zip) [ext-imagick-php8.2-nts-x64.zip](https://github.com/Imagick/imagick/files/10758244/ext-imagick-php8.2-nts-x64.zip)

PHP 8.2.3 x64 TS 

Works for me, W11 + XAMPP with PHP 8.2, Thanks!

**Comment by Qara-Penimah** [comment]
Is there an official release?

**Comment by N6REJ** [comment]
when I compare the ts to whats on pecl repo there is a LOT missing.  I'm unsure what to do.

**Comment by WilliamStam** [comment]
it works for everything i need it to do. doesnt error out. are there any issues with the differences? otherwise i wouldnt worry too much i think

**Comment by ArthurPatriot** [comment]
> TS here so that might be less of an issue than TS version i suppose

all must works

**Comment by Asmitta-01** [comment]
> Success BUILD for Windows! Test on your own risk. Left messages here if it works.
> 
> PHP 8.2.3 x64 TS + NTS
> 
> [ext-imagick-php8.2-ts-x64.zip](https://github.com/Imagick/imagick/files/10745985/ext-imagick-php8.2-ts-x64.zip) [ext-imagick-php8.2-nts-x64.zip](https://github.com/Imagick/imagick/files/10758244/ext-imagick-php8.2-nts-x64.zip)

It's working. Thanks

**Comment by ELFswe** [comment]
@ArthurPatriot
Any possibillity of a lighter guide for wich rings of hell you went through to build this?

**Comment by ArthurPatriot** [comment]
> @ArthurPatriot Any possibillity of a lighter guide for wich rings of hell you went through to build this?

In fact, it is difficult only the first time... Once you have everything figured out, it is done in a few minutes...

**Comment by jamesjohnmcguire** [comment]
@ArthurPatriot Would it possible to publish some notes on how to build this?  Even if general, could be helpful.  But detailed, step by step, instructions, or even a build script would be great!

For those of who also want to build our own.

**Comment by fmonts** [comment]
> ImageMagick number of supported formats:  => 0

I have the same problem with 0 supported formats, on multiple machines... anybody managed to fix this?

**Comment by fmonts** [comment]
I found a working solution:

- 8.2 binaries from the zip above (7.1.0-62) don't work (got 0 supported formats); probably something is broken in the build
BUT
- 8.1 binaries, the official ones from [here](https://windows.php.net/downloads/pecl/releases/imagick/3.7.0/) do work on 8.2 as well. You only need php_imagick.dll build for 8.2 from the zip above.

so:
- in ext/, put the .dll for 8.2
- in bin/, put the binaries for 8.1

Tested this setup on 3 machines in our team, on both windows 10 and windows 11

**Comment by N6REJ** [comment]
can we get this into a working windows zip like the existing?

**Comment by N6REJ** [comment]
how about we get an "official" fork so we can keep up to date instead of a long thread

**Comment by Nomad862** [comment]
bonjour à tous, pour le problème de imagick j'ai trouvé la solution après plusieurs essaie et recherche. 
Voila de 1 je ne suis ni dévelloppeur ni programmeur et je n'est aucune connaissance ni en dévellopement ni dans les codes mais je ne suis pas con hhhhhhhh de 2 je travail sur XAMPP pour info 
en ce qui concerne la version PHP elle est de 8.2
J'ai commencer par installe l'application "ImageMagick-7.1.1-12-Q16-HDRI-x64-dll" dans mon systeme
Ensuite j'ai télécharger la derniere version de php image magick "php_imagick-3.7.0-8.1-ts-vs16-x64" ou "php_imagick-3.7.0-8.1-nts-vs16-x64" et j'ai extrait les fichiers afin d'avoir le dossier concernant
Apres j'ai copie le fichier     "php_imagick.dll"      dans le dossier XAMPP/php/ext
ensuite j'ai copie tous les fichiers qui commence par "CORE_........_.dll" et les fichiers "IM_MOD_......._.dll"    dans le dossier XAMPP/Apache/bin
 enfin j'ai copié le fichier "extension=php_imagick.dll" sans la virgule avec point dans le fichier PHP.INI au dessus des extension ";extension=pgsql    ///  ;extension=shmop"
Ca na pas fonctionner, mais quand j'ai ajouter le fichier "extension=php_gd2.dll" dans le fichier PHP.INI ça à fonctionner je n'est plus le message du probleme du imagick ni du GD 
si ça ne fonctionne pas verifier si le fichier "php_gd.dll" existe dans le dossier XAMPP/apache/bin si oui je suis désolé que ça ne marche pas pour vous si non copie "php_gd.dll" dans le fichier XAMPP/apache/bin

**Comment by ferrisbuellers** [comment]
What's holding this up from being officially released on Windows PHP8.2?

**Comment by ELFswe** [comment]
> What's holding this up from being officially released on Windows PHP8.2?
There has never really been any official releases for Windows.
https://github.com/Imagick/imagick/issues/573#issuecomment-1353609195

**Comment by terax6669** [comment]
> There has never really been any official releases for Windows.

The DLLs are available for download on https://pecl.php.net/package/imagick, does it get more official than that?

**Comment by WilliamStam** [comment]
> The DLLs are available for download on https://pecl.php.net/package/imagick, does it get more official than that?

the issue here is that https://pecl.php.net/package/imagick/3.7.0/windows only has for php 8.1. and the dll doesnt work on php 8.2. so a rather weird post.

**Comment by ArthurPatriot** [comment]
> > The DLLs are available for download on https://pecl.php.net/package/imagick, does it get more official than that?
> 
> the issue here is that https://pecl.php.net/package/imagick/3.7.0/windows only has for php 8.1. and the dll doesnt work on php 8.2. so a rather weird post.

https://github.com/Imagick/imagick/issues/573#issuecomment-1431773928

**Comment by WilliamStam** [comment]
> [#573 (comment)](https://github.com/Imagick/imagick/issues/573#issuecomment-1431773928)

im already using that and it works perfectly :D my comment was aimed at the person that said that pecl = official, was just commenting that there isnt 8.2 dlls there so nothing "official"

**Comment by Danack** [comment]
> but I believe the Windows builds are generated automatically by PECL.

No, the box died, and previously it was sponsored by Microsoft. Microsoft pulled out of helping the PHP project. So there is nothing I can do myself.

The PHP core team is really low on members, and it's 'not obvious' that getting volunteers to step in to support an OS after the company selling it drops contributing, is the right thing to do.

**Comment by EusebiuOprinoiu** [comment]
That's disappointing.
I remember seeing an announcement about the build box dying about a year ago, but I thought it was fixed.
At least the Linux version works with all PHP versions, even though it's a bit old.

Will you be able to release a new version compiled with ImageMagick 7 in the near future?
To be clear, I'm asking about the Linux version available via PECL which is used by many install scripts.

**Comment by ArthurPatriot** [comment]
> That's disappointing. I remember seeing an announcement about the build box dying about a year ago, but I thought it was fixed. At least the Linux version works with all PHP versions, even though it's a bit old.
> 
> Will you be able to release a new version compiled with ImageMagick 7 in the near future? To be clear, I'm asking about the Linux version available via PECL which is used by many install scripts.

This issue about Windows build. Linux versions working well.

**Comment by mcbmcb0** [comment]
as this appears to be the current php 8.2 thread...
I running php 8.2.4 on win 10 (xampp) and doing this as suggested by @fmonts:
` in ext/, put the .dll for 8.2`
` in bin/ (or whatever folder you added to windows Path), put the binaries for 8.1`
then  `php --ri imagick` shows imagick module enabled and 263 supported formats.
But the xampp apache service says php_imagick.dll is not found. It is there and I've tried locating it from php.ini in a number of ways.
can someone confirm if they have got this running with php 8.2.4 with Apache 2.4.56?

thanks

**Comment by adamperrry** [comment]
@ArthurPatriot 

I've noticed the module version is coming back as `@PACKAGE_VERSION`, which is breaking PDF previews in Wordpress media uploads, because the `class-wp-image-editor-imagick.php` class has a `test()` function that contains:

```php
		if ( version_compare( phpversion( 'imagick' ), '2.2.0', '<' ) ) {
			return false;
		}
```

I've got the 8.1 ImageMagick binaries, so everything is configured and working correctly, except the version. Is this an issue with the compilation process?

```bash
$ php --ri imagick

imagick

imagick module => enabled
imagick module version => @PACKAGE_VERSION@
imagick classes => Imagick, ImagickDraw, ImagickPixel, ImagickPixelIterator, ImagickKernel
Imagick compiled with ImageMagick version => ImageMagick 7.1.0-62 Q16-HDRI x64 32ce406:20230212 https://imagemagick.org
Imagick using ImageMagick library version => ImageMagick 7.1.0-18 Q16 x64 2021-12-14 https://imagemagick.org
ImageMagick copyright => (C) 1999-2021 ImageMagick Studio LLC
ImageMagick release date => 2021-12-14
ImageMagick number of supported formats:  => 263
```

**Comment by ArthurPatriot** [comment]
> @ArthurPatriot
> 
> I've noticed the module version is coming back as `@PACKAGE_VERSION@`, which is breaking PDF previews in Wordpress media uploads, because the `class-wp-image-editor-imagick.php` class has a `test()` function that contains:
> 
> ```
> 		if ( version_compare( phpversion( 'imagick' ), '2.2.0', '<' ) ) {
> 			return false;
> 		}
> ```
> 
> I've got the 8.1 ImageMagick binaries, so everything is configured and working correctly, except the version. Is this an issue with the compilation process?
> 
> ```shell
> $ php --ri imagick
> 
> imagick
> 
> imagick module => enabled
> imagick module version => @PACKAGE_VERSION@
> imagick classes => Imagick, ImagickDraw, ImagickPixel, ImagickPixelIterator, ImagickKernel
> Imagick compiled with ImageMagick version => ImageMagick 7.1.0-62 Q16-HDRI x64 32ce406:20230212 https://imagemagick.org
> Imagick using ImageMagick library version => ImageMagick 7.1.0-18 Q16 x64 2021-12-14 https://imagemagick.org
> ImageMagick copyright => (C) 1999-2021 ImageMagick Studio LLC
> ImageMagick release date => 2021-12-14
> ImageMagick number of supported formats:  => 263
> ```

To fix this issue extension must be rebuilded with filled module version...

**Comment by N6REJ** [comment]
using http://bearsampp.com I don't have that problem.
<img width="989" alt="image" src="https://github.com/Imagick/imagick/assets/1850089/afdbbdbc-f54b-4d7c-aaea-24a1162f44c0">

**Comment by N6REJ** [comment]
> as this appears to be the current windows php 8.2 thread... I running php 8.2.4 on win 10 (xampp) and doing this as suggested by @fmonts: ` in ext/, put the .dll for 8.2` ` in bin/ (or whatever folder you added to windows Path), put the binaries for 8.1` then `php --ri imagick` shows imagick module enabled and 263 supported formats. But the xampp apache service says php_imagick.dll is not found. It is there and I've tried locating it from php.ini in a number of ways. can someone confirm if they have got this running on windows with php 8.2.4 with Apache 2.4.56? thanks
> 
> **EDIT:** I translated the comment from @Nomad862 above and so also copied the 8.1 CORE_.........dll" and the "IM_MOD......._.dll" files into the XAMPP/Apache/bin folder. They are also in ..xampp/php/ Imagick now appears to work with apache / php 8.2.4 / win 10. i'm not sure if my installation was somehow broken or some other problem, but problem appears solved

I can confirm 8.2 doesn't work properly
```
> php --ri imagick
PHP Warning:  PHP Startup: imagick: Unable to initialize module
Module compiled with module API=20210902
PHP    compiled with module API=20220829
These options need to match
 in Unknown on line 0

Warning: PHP Startup: imagick: Unable to initialize module
Module compiled with module API=20210902
PHP    compiled with module API=20220829
These options need to match
 in Unknown on line 0
Extension 'imagick' not present.
```

**Comment by ArthurPatriot** [comment]
> @ArthurPatriot using http://bearsampp.com I don't have that problem. <img alt="image" width="989" src="https://user-images.githubusercontent.com/1850089/260842142-afdbbdbc-f54b-4d7c-aaea-24a1162f44c0.png">

Issue with 8.2

**Comment by Asmitta-01** [comment]
Is there an update for this package coming for php 8.2 ? 😔

**Comment by ArthurPatriot** [comment]
> Is there an update for this package coming for php 8.2 ? 😔

https://github.com/Imagick/imagick/issues/573#issuecomment-1431773928

**Comment by N6REJ** [comment]
> ImageMagick-7.1.0-62-Q16-HDRI-x64

we have to use https://imagemagick.org/archive/binaries/ImageMagick-7.1.1-15-portable-Q16-x64.zip, as far as i know, in https://github.com/Bearsampp/Bearsampp due to it being 100% portable?  by doing so it doesn't work.  We use ant to create the php module so I'm not sure what exactly happens when its run.  Here's our 8.2.8 module https://github.com/Bearsampp/module-php/tree/main/bin/php8.2.8

**Comment by Duber79** [comment]
> Success BUILD for Windows! Test on your own risk. Left messages here if it works.
> 
> PHP 8.2.3 x64 TS + NTS
> 
> [ext-imagick-php8.2-ts-x64.zip](https://github.com/Imagick/imagick/files/10745985/ext-imagick-php8.2-ts-x64.zip) [ext-imagick-php8.2-nts-x64.zip](https://github.com/Imagick/imagick/files/10758244/ext-imagick-php8.2-nts-x64.zip)

Hello. Please Rebuild the dll extension for the version ImageMagick-7.1.1-15-Q16-HDRI-x64-dll

**Comment by N6REJ** [comment]
we're getting a bug when using ... 
what is the proper build for 8.2.x and is there a repo we can follow for the latest builds?  This is getting huge and hard to follow.
<img width="685" alt="image" src="https://github.com/Imagick/imagick/assets/1850089/1f031de8-3add-4d1b-ac4d-eeae163bef05">

**Comment by Geckosan** [comment]
Hi, I've done a recent imagick build for php 8.2 windows x64 after I had issues getting the previous build posted here from working (0 supported file types, mismatched versions etc), I'm attaching here in case its useful to anybody.
Obviously, use at your own risk.

Built using PHP 8.2.11 vs16 x64, ImageMagick 7.1.1-20 (Beta) Q16-HDRI vs16 x64 (Dynamic Multi-threaded DLL version), and imagick 3.7.0.

I've set the module version to 3.7.0 (which looking at the source I'm probably not supposed to do) so stuff that checks the version should be ok.

Both TS and NTS versions are included (just copy the right version of the dll to your php/ext folder and rename it to php_imagick.dll). Add the unzipped folder to your windows system path and you should be good to go.
![imagick-php8211-phpinfo](https://github.com/Imagick/imagick/assets/33095644/0c2ed535-daa2-40f4-a189-9637480343c7)
[php_imagick_3.7.0-php_8.2-vs16-x64.zip](https://github.com/Imagick/imagick/files/12806169/php_imagick_3.7.0-php_8.2-vs16-x64.zip)

**Comment by N6REJ** [comment]
Can we get a repo of our own?  Would be great to be able to use the release features instead of going thru chat

**Comment by ArthurPatriot** [comment]
> Can we get a repo of our own? Would be great to be able to use the release features instead of going thru chat

I could make a service that automatically builds new versions of extensions for php, but I'm too lazy to do it😅

**Comment by N6REJ** [comment]
you'd have a very grateful bear if you did!  we use it in our development server we offer @ http://bearsampp.com

**Comment by ArthurPatriot** [comment]
> you'd have a very grateful bear if you did! we use it in our development server we offer @ http://bearsampp.com

I will think about it and maybe start work in this direction

**Comment by zavsc** [comment]
> Hi, I've done a recent imagick build for php 8.2 windows x64 after I had issues getting the previous build posted here from working (0 supported file types, mismatched versions etc), I'm attaching here in case its useful to anybody. Obviously, use at your own risk.
> 
> Built using PHP 8.2.11 vs16 x64, ImageMagick 7.1.1-19 Q16-HDRI vs16 x64 (Dynamic Multi-threaded DLL version), and imagick 3.7.0.
> 
> I've set the module version to 3.7.0 (which looking at the source I'm probably not supposed to do) so stuff that checks the version should be ok.
> 
> Both TS and NTS versions are included (just copy the right version of the dll to your php/ext folder and rename it to php_imagick.dll). Add the unzipped folder to your windows system path and you should be good to go.
> 
> **Edit** dropped ImageMagick version to 7.1.1-19 instead of the beta version. ![imagick-php8211-phpinfo](https://user-images.githubusercontent.com/33095644/272918193-d441b27c-8c2a-41b6-9bb8-5eab967cc493.jpg) [php_imagick_3.7.0-php_8.2-vs16-x64.zip](https://github.com/Imagick/imagick/files/12818890/php_imagick_3.7.0-php_8.2-vs16-x64.zip)

Can you build imagick for PHP 8.2.12 NTS?

**Comment by Geckosan** [comment]
> Can you build imagick for PHP 8.2.12 NTS?

The extension compiled for PHP 8.2.11 should work with any 8.2.* release.

**Comment by mudcamper** [comment]
Thank you Geckosan! It works for me.

**Comment by zhoumann** [comment]
Thank you so much for this Geckosan! Will you be compiling one for PHP 8.3, as it's officially released now?

**Comment by Geckosan** [comment]
> Thank you so much for this Geckosan! Will you be compiling one for PHP 8.3, as it's officially released now?

I'll take a look next week

**Comment by Geckosan** [comment]
Provisional build for windows PHP 8.3. Both TS & NTS versions are included in the zip.
I'm not in a position check this right now, so please test thoroughly before use.

Unzip to a location of your choice. Add the unzipped directory to your windows system path (make sure it comes before any previously installed versions of ImageMagick). Copy either php_imagick_ts.dll or php_imagick_nts.dll (depending on your PHP version) to your php/ext directory and rename to php_imagick.dll. Enable the extension in php.ini and run phpinfo to check it's working.
[php_imagick_3.7.0-php_8.3-vs16-x64.zip](https://github.com/Imagick/imagick/files/13473323/php_imagick_3.7.0-php_8.3-vs16-x64.zip)

**Comment by zhoumann** [comment]
> Provisional build for windows PHP 8.3. Both TS & NTS versions are included in the zip. I'm not in a position check this right now, so please test thoroughly before use.
> 
> Unzip to a location of your choice. Add the unzipped directory to your windows system path (make sure it comes before any previously installed versions of ImageMagick). Copy either php_imagick_ts.dll or php_imagick_nts.dll (depending on your PHP version) to your php/ext directory and rename to php_imagick.dll. Enable the extension in php.ini and run phpinfo to check it's working. [php_imagick_3.7.0-php_8.3-vs16-x64.zip](https://github.com/Imagick/imagick/files/13473323/php_imagick_3.7.0-php_8.3-vs16-x64.zip)

Thank you so much Geckosan, I've installed it on my Windows machine and it works perfectly! (I've only tested the thread-safe version)

![Web capture_27-11-2023_125429_localhost](https://github.com/Imagick/imagick/assets/5460401/6722a606-d764-4843-bcb0-d270d357cdcf)

**Comment by Duber79** [comment]
Please upload the installer ImageMagick-7.1.1-19-Q16-HDRI-x64-dll.exe, otherwise it won't download from the official website.

**Comment by bally86** [comment]
Where can I find the ImageMagick 7.1.1-19 Q16-HDRI x64 installer?

**Comment by danielscsc** [comment]
> Where can I find the ImageMagick 7.1.1-19 Q16-HDRI x64 installer?

I'm not sure you need that exact version, but if you do:
https://imagemagick.org/archive/binaries/

**Comment by danielscsc** [comment]
> > > Where can I find the ImageMagick 7.1.1-19 Q16-HDRI x64 installer?
> > 
> > 
> > I'm not sure you need that exact version, but if you do: https://imagemagick.org/archive/binaries/
> 
> I only see 7.1.1-25 there. Can I just use that one?

I would think so.  Can't hurt to try :)  Let us know!

**Comment by Duber79** [comment]
It didn't work for me, PHP 8.2.9 (NTS).
Error loading the module(

**Comment by joakim-tldr** [comment]
> Hi, I've done a recent imagick build for php 8.2 windows x64 after I had issues getting the previous build posted here from working (0 supported file types, mismatched versions etc), I'm attaching here in case its useful to anybody. Obviously, use at your own risk.
> 
> Built using PHP 8.2.11 vs16 x64, ImageMagick 7.1.1-19 Q16-HDRI vs16 x64 (Dynamic Multi-threaded DLL version), and imagick 3.7.0.
> 
> I've set the module version to 3.7.0 (which looking at the source I'm probably not supposed to do) so stuff that checks the version should be ok.
> 
> Both TS and NTS versions are included (just copy the right version of the dll to your php/ext folder and rename it to php_imagick.dll). Add the unzipped folder to your windows system path and you should be good to go.
> 
> **Edit** dropped ImageMagick version to 7.1.1-19 instead of the beta version. ![imagick-php8211-phpinfo](https://private-user-images.githubusercontent.com/33095644/272918193-d441b27c-8c2a-41b6-9bb8-5eab967cc493.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY0NDgxODksIm5iZiI6MTcwNjQ0Nzg4OSwicGF0aCI6Ii8zMzA5NTY0NC8yNzI5MTgxOTMtZDQ0MWIyN2MtOGMyYS00MWI2LTliYjgtNWVhYjk2N2NjNDkzLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI4VDEzMTgwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3YmQ0ZjEyOGIzYzgzNTliYjBkZTJjN2U0Y2IwYTlhZTZmZDMyZjRkMjA2ZDUzZWI5ODk1ZTFlYmRlOGM0N2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g6NmgYi6EJl1kNmMU9pBz9e8y5P9KYZuDKsCSUxGDU4) [php_imagick_3.7.0-php_8.2-vs16-x64.zip](https://github.com/Imagick/imagick/files/12818890/php_imagick_3.7.0-php_8.2-vs16-x64.zip)

Thank you! Can confirm php_imagick_ts work for any PHP 8.2.* with ImageMagick 7.1.1-19 Q16-HDRI vs16 x64 (Dynamic Multi-threaded DLL version), and imagick 3.7.0, added it to several machines.

**Comment by christoferd** [comment]
Is there a dll for imagick for PHP 8.24 VS16 x64 TS?
Or, any of the later PHP 8 releases?

**Comment by lija0905** [comment]
Bless this thread. I've lost so many hours on this installation. Only information from here helped me!!!!

**Comment by ayuniptrii2** [comment]
> Success BUILD for Windows! Test on your own risk. Left messages here if it works.
> 
> PHP 8.2.3 x64 TS + NTS
> 
> [ext-imagick-php8.2-ts-x64.zip](https://github.com/Imagick/imagick/files/10745985/ext-imagick-php8.2-ts-x64.zip) [ext-imagick-php8.2-nts-x64.zip](https://github.com/Imagick/imagick/files/10758244/ext-imagick-php8.2-nts-x64.zip)

It works. thanks. i've searched the pecl web but it's still up to php 8.1

**Comment by ben-norton** [comment]
PHP 8.3 version?

**Comment by johnzoet** [comment]
> Success BUILD for Windows! Test on your own risk. Left messages here if it works.
> 
> PHP 8.2.3 x64 TS + NTS
> 
> [ext-imagick-php8.2-ts-x64.zip](https://github.com/Imagick/imagick/files/10745985/ext-imagick-php8.2-ts-x64.zip) [ext-imagick-php8.2-nts-x64.zip](https://github.com/Imagick/imagick/files/10758244/ext-imagick-php8.2-nts-x64.zip)

Working for me.
Installed ImageMagick DLL with higher version. Tested and working.
Further, I copied all DLL files from imagick zip file to a folder and added that folder to the system environment path.
One caveat I encountered is that phpinfo.php did not have ImageMagick on the system path. Restarting Xampp solved this problem.

**Comment by N6REJ** [comment]
I'd like to offer to make a repo where y'all can contribute and we can actually have releases.  I think it would be good to have imagemagick and imagick both under one repo so we can keep things up to date and easy fo find.  I'd like y'alls feedback on this.

**Comment by Rentner2** [comment]
I think it's a good idea. Go on!

**Comment by ArthurPatriot** [comment]
> I'd like to offer to make a repo where y'all can contribute and we can actually have releases. I think it would be good to have imagemagick and imagick both under one repo so we can keep things up to date and easy to find. I'd like y'alls feedback on this. We're posting releases in this thread but not under releases which makes them hard to find.

PECL windows builder is broken. So you will get same story for all extensions. 

I am planning to write simple service which build required extension for requested php version... On same way as pecl. What you think guys?)

**Comment by N6REJ** [comment]
> > I'd like to offer to make a repo where y'all can contribute and we can actually have releases. I think it would be good to have imagemagick and imagick both under one repo so we can keep things up to date and easy to find. I'd like y'alls feedback on this. We're posting releases in this thread but not under releases which makes them hard to find.
> 
> PECL windows builder is broken. So you will get same story for all extensions.
> 
> I am planning to write simple service which build required extension for requested php version... On same way as pecl. What you think guys?)

PLEASE DO!!!

**Comment by ArthurPatriot** [comment]
@N6REJ Look like already done - https://phpext.phptools.online/extension/images/imagick-121

**Comment by ArthurPatriot** [comment]
> PHP 8.3 version?

https://phpext.phptools.online/extension/images/imagick-121

**Comment by ArthurPatriot** [comment]
> Is there a dll for imagick for PHP 8.24 VS16 x64 TS? Or, any of the later PHP 8 releases?

https://phpext.phptools.online/extension/images/imagick-121

**Comment by CopaceticMeatbag** [comment]
> Provisional build for windows PHP 8.3. Both TS & NTS versions are included in the zip. I'm not in a position check this right now, so please test thoroughly before use.
> 
> Unzip to a location of your choice. Add the unzipped directory to your windows system path (make sure it comes before any previously installed versions of ImageMagick). Copy either php_imagick_ts.dll or php_imagick_nts.dll (depending on your PHP version) to your php/ext directory and rename to php_imagick.dll. Enable the extension in php.ini and run phpinfo to check it's working. [php_imagick_3.7.0-php_8.3-vs16-x64.zip](https://github.com/Imagick/imagick/files/13473323/php_imagick_3.7.0-php_8.3-vs16-x64.zip)

Confirmed this worked without any issues using the latest "7.1.1-32 Q16-HDRI vs16 x64" installer and PHP 8.3.6. Many thanks!!

**Comment by fmonts** [comment]
It seems the official PECLs for 8.2 and 8.3 are now released

https://mlocati.github.io/articles/php-windows-imagick.html

**Comment by EusebiuOprinoiu** [comment]
Finally!
Unfortunately, only the Windows builds have been updated.
The Linux package is still the old version from January 2022, compiled with ImageMagick 6.9.

I made a ticket about this about a year ago if you want to add your voice!
https://github.com/Imagick/imagick/issues/625

**Comment by numen111104** [comment]
> Provisional build for windows PHP 8.3. Both TS & NTS versions are included in the zip. I'm not in a position check this right now, so please test thoroughly before use.
> 
> Unzip to a location of your choice. Add the unzipped directory to your windows system path (make sure it comes before any previously installed versions of ImageMagick). Copy either php_imagick_ts.dll or php_imagick_nts.dll (depending on your PHP version) to your php/ext directory and rename to php_imagick.dll. Enable the extension in php.ini and run phpinfo to check it's working. [php_imagick_3.7.0-php_8.3-vs16-x64.zip](https://github.com/Imagick/imagick/files/13473323/php_imagick_3.7.0-php_8.3-vs16-x64.zip)

THANKSS SO MUCCH IT'S WORK TO MEEE

**Comment by zhoumann** [comment]
Any chance of a PHP 8.4 build? Not sure when/if the official PECL package is going to have a PHP 8.4 DLL hook for imagick.

**Comment by fmonts** [comment]
> Any chance of a PHP 8.4 build? Not sure when/if the official PECL package is going to have a PHP 8.4 DLL hook for imagick.

8.4 official PECL is available! 

https://pecl.php.net/package/imagick/3.7.0/windows 

Working fine to me

**Comment by bally86** [comment]
> > Any chance of a PHP 8.4 build? Not sure when/if the official PECL package is going to have a PHP 8.4 DLL hook for imagick.
> 
> 8.4 official PECL is available!
> 
> https://pecl.php.net/package/imagick/3.7.0/windows
> 
> Working fine to me

I downloaded this version, but when I run php --version it returns:

Warning: Version warning: Imagick was compiled against ImageMagick version 1808 but version 1809 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0
PHP 8.4.3 (cli) (built: Jan 15 2025 11:02:41) (NTS Visual C++ 2022 x64)

Do I need a other version of ImageMagick? Currently i'm using:
ImageMagick 7.1.1-43 Q16-HDRI x64 a2d96f4:20241222

**Comment by bukka** [comment]
The official Windows builds for the latest PHP versions and ImageMagick 8.1.1-46 have just been released and are available at https://github.com/Imagick/imagick/releases/tag/3.8.0
